silu_and_mul kernel only supports contiguous tensors

mwbyeon opened the pull request:
`ops.silu_and_mul` and `ops.gelu_and_mul` kernel only supports contiguous tensors.

Non-contiguous tensors can also be run, but they will output incorrect results and fail unit tests. 
Therefore, when calling the `silu_and_mul` and `gelu_and_mul` kernels, modify them to use contiguous tensors.

mwbyeon committed:
Fixed bug in ops.silu_and_mul kernel when input is not contiguous

mwbyeon committed:
Merge branch 'main' into bugfix/act_and_mul_noncontiguous

WoosukKwon commented:
Hi @mwbyeon, thanks for submitting the PR! Could you please provide an example where the input tensors might be non-contiguous? We haven't encountered this scenario before.

mwbyeon commented:
@WoosukKwon 

It happened while trying to implement parallel layers of MHA and MLP layers, which are used in CodeGen, PaLM, GPT-NeoX, and Phi papers. 

Please see https://hpc-ai.com/blog/implementing-the-gargantuan-pathways-with-colossal-ai-easy-and-efficient
I extended the `QKVParallelLinear` to implement Figure 2. When splitting the "Fused Tensor" in two (`tensor.split`) and using it as input to the MLP and Attention modules respectively, the "MLP Tensor" became non-contiguous.

There are two solutions to this situation
1. use `self.mlp(x.contiguous())` in the Decoder Layer
2. use `ops.silu_and_mul(x.contiguous())` as in this PR


I implemented it like solution 2 because I think it could happen in other situations where `ops.silu_and_mul` is used, 
but solution 1 is also a good.
